### PR TITLE
Fix rest API test timeout

### LINKED ISSUE #1510: REST API tests timeout in CircleCI
**Detailed Description**
REST API integration tests randomly timeout in CircleCI. It's not really specific to a single test.

- https://app.circleci.com/pipelines/github/hashgraph/hedera-mirror-node/5633/workflows/7b1a6b19-dd86-42fa-9343-4b1104ee3684/jobs/16086/steps
- https://app.circleci.com/pipelines/github/hashgraph/hedera-mirror-node/5628/workflows/d7148652-c68f-446b-883b-591bbeeb5ee0/jobs/16072/steps
- https://app.circleci.com/pipelines/github/hashgraph/hedera-mirror-node/5626/workflows/2a5a2bbe-87f6-41fb-b6ab-c6e7d3c11504/jobs/16067/steps
- https://app.circleci.com/pipelines/github/hashgraph/hedera-mirror-node/5626/workflows/698c09d9-b763-4b21-942c-f1159f19cb27/jobs/16064
- https://app.circleci.com/pipelines/github/hashgraph/hedera-mirror-node/5570/workflows/412de44e-3043-4508-a632-16b8f9a6b93d/jobs/15903

**Actual Behavior**
Sampling of errors:
```
  ● Balances tests › Balances single test: timestamp_lowerlimit - URL: timestamp=gte:1611786768

    : Timeout - Async callback was not invoked within the 1000 ms timeout specified by jest.setTimeout.Timeout - Async callback was not invoked within the 1000 ms timeout specified by jest.setTimeout.Error:

      290 |   // First, execute the single tests
      291 |   for (const [name, item] of Object.entries(singletests)) {
    > 292 |     test(`Balances single test: ${name} - URL: ${item.urlparam}`, async () => {
          |     ^
      293 |       let response = await request(server).get([api, item.urlparam].join('?'));
      294 | 
      295 |       expect(response.status).toEqual(200);

      at new Spec (node_modules/jest-jasmine2/build/jasmine/Spec.js:116:22)
      at Suite.<anonymous> (__tests__/balances.test.js:292:5)
      at Object.<anonymous> (__tests__/balances.test.js:287:1)
```

```
Accounts tests › Accounts single test: accountid_lowerlimit - URL: account.id=gte:0.0.1111

    : Timeout - Async callback was not invoked within the 1000 ms timeout specified by jest.setTimeout.Timeout - Async callback was not invoked within the 1000 ms timeout specified by jest.setTimeout.Error:

      245 |   // First, execute the single tests
      246 |   for (const [name, item] of Object.entries(singletests)) {
    > 247 |     test(`Accounts single test: ${name} - URL: ${item.urlparam}`, async () => {
          |     ^
      248 |       let response = await request(server).get([api, item.urlparam].join('?'));
      249 | 
      250 |       expect(response.status).toEqual(200);

      at new Spec (node_modules/jest-jasmine2/build/jasmine/Spec.js:116:22)
      at Suite.<anonymous> (__tests__/accounts.test.js:247:5)
      at Object.<anonymous> (__tests__/accounts.test.js:242:1)

  ● Accounts tests › Accounts single test: accountpublickey_equal - URL: account.publickey=6bd7b31fd59fc1b51314ac90253dfdbffa18eec48c00051e92635fe964a08c9b

    : Timeout - Async callback was not invoked within the 1000 ms timeout specified by jest.setTimeout.Timeout - Async callback was not invoked within the 1000 ms timeout specified by jest.setTimeout.Error:

      245 |   // First, execute the single tests
      246 |   for (const [name, item] of Object.entries(singletests)) {
    > 247 |     test(`Accounts single test: ${name} - URL: ${item.urlparam}`, async () => {
          |     ^
      248 |       let response = await request(server).get([api, item.urlparam].join('?'));
      249 | 
      250 |       expect(response.status).toEqual(200);

      at new Spec (node_modules/jest-jasmine2/build/jasmine/Spec.js:116:22)
      at Suite.<anonymous> (__tests__/accounts.test.js:247:5)
      at Object.<anonymous> (__tests__/accounts.test.js:242:1)

  ● Accounts tests › Accounts single test: limit - URL: limit=99

    : Timeout - Async callback was not invoked within the 1000 ms timeout specified by jest.setTimeout.Timeout - Async callback was not invoked within the 1000 ms timeout specified by jest.setTimeout.Error:

      245 |   // First, execute the single tests
      246 |   for (const [name, item] of Object.entries(singletests)) {
    > 247 |     test(`Accounts single test: ${name} - URL: ${item.urlparam}`, async () => {
          |     ^
      248 |       let response = await request(server).get([api, item.urlparam].join('?'));
      249 | 
      250 |       expect(response.status).toEqual(200);

      at new Spec (node_modules/jest-jasmine2/build/jasmine/Spec.js:116:22)
      at Suite.<anonymous> (__tests__/accounts.test.js:247:5)
      at Object.<anonymous> (__tests__/accounts.test.js:242:1)
```

**Expected Behavior**
Test to pass

**Environment:**
- Version: master

**Additional Context**

### PR DESCRIPTION
**Pull request by xin-hedera**
**Detailed description**:
- Set default jest timeout to 2000ms
- Move `jest.setTimeout` out of before* hook in `integration.test.js` as it does not work with `jest-circus` test runner
- Change jest `testEnvironment` to `node`
- Disable `metrics` during test as it causes open handle with `node` testEnvironment
- Use awaitjs express `useAsync` for async middleware

**Which issue(s) this PR fixes**:
Fixes #1510 

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [x] Tests updated

### CODE REVIEW

File: hedera-mirror-rest/middleware/httpErrorHandler.js [hunk 1]
```diff
@@ -59,14 +59,13 @@ const handleError = async (err, req, res, next) => {
 
   if (shouldReturnMessage(statusCode)) {
     errorMessage = err.message;
-    logger.warn(`${statusCode} processing ${req.originalUrl}: ${err.constructor.name} ${errorMessage}`);
+    logger.warn(`${statusCode} processing ${req.originalUrl}: `, err);
```
Comment (steven-sheehy) [inline hedera-mirror-rest/middleware/httpErrorHandler.js:62]: This change was intentional. We don't need to print a full stack trace for user caused errors like invalid input. Stacktraces are for things we need to potentially investigate.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: hedera-mirror-rest/middleware/httpErrorHandler.js (hunk 1 above)
Comment (xin-hedera) [reply hedera-mirror-rest/middleware/httpErrorHandler.js:62]: will revert. It was changed to check where the `NotFound` error was from for the issue with 0.28.0-beta1

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Review by Nana-EC — APPROVED** [review]
LGTM

**Pull request merged**